File: Resource/51900286-51900712-51900718-51900699/source demo/Web Application/ComNha_API/ComNha_API/Models/Khachhang.cs
using System;
using System.Collections.Generic;

namespace ComNha_API.Models
{
    public partial class Khachhang
    {
        public Khachhang()
        {
            Hoadons = new HashSet<Hoadon>();
            Lichhens = new HashSet<Lichhen>();
        }

        public int MaKhachHang { get; set; }
        public string SoDienThoai { get; set; } = null!;
        public string TenKhachHang { get; set; } = null!;
        public string? DiaChi { get; set; }
        public string? Email { get; set; }

        public virtual ICollection<Hoadon> Hoadons { get; set; }
        public virtual ICollection<Lichhen> Lichhens { get; set; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/KhachHang.java
package tdtu.advanced.java.thinh68.models;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name="khach_hang")
public class KhachHang {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    @Column(name="ma_khach_hang")
    private Long maKhachHang;
    
    @Column(name="so_dien_thoai", nullable = false)
    private String soDienThoai;
    
    @Column(name="ten_khach_hang", nullable = false)
    private String tenKhachHang;
    
    @Column(name="dia_chi", nullable = true)
    private String diaChi;
    
    @Column(name="email", nullable = true)
    private String email;
    

    @OneToMany(mappedBy="khachHang")
    @JsonIgnore
    private List<LichHen> khachHang_LichHens;
    

    @OneToMany(mappedBy="khachHang")
    @JsonIgnore
    private List<HoaDon> khachHang_HoaDons;
}